pkg/userdata/amzn2/provider.go

```diff
@@ -300,6 +300,18 @@ write_files:
   content: |
 {{ kubeletHealthCheckSystemdUnit | indent 4 }}
 
+{{- with .ProviderSpec.CAPublicKey }}
+
```

Comment:
Is this newline intentional?

Reply:
I saw the rendered version and looks good, probably you could have skipped the dash and the newline:

```
{{ with .ProviderSpec.CAPublicKey }}
- path: "/etc/ssh/trusted-user-ca-keys.pem"
```
but it's ok.

Reply:
Hello

It was intentional as the next block would fall on the next line with this so the spacing is to accomodate for legibility

```yaml
- path:dukfl;d
  key: value and some other content
- path: huf
```

instead of 
```yaml
- path:dukfl;d
  key: value and some other content
  
- path: huf
```

Nothing more